src/managed/Microsoft.Extensions.DependencyModel/UnifiedJsonReader.Utf8JsonReader.cs

```diff
@@ -96,21 +96,7 @@ public string[] ReadStringArray()
             return items.ToArray();
         }
 
-        public void Skip()
-        {
-            if (IsTokenTypeProperty())
-            {
-                _reader.Read();
-            }
-
-            if (_reader.TokenType == JsonTokenType.StartObject || _reader.TokenType == JsonTokenType.StartArray)
-            {
-                int depth = _reader.CurrentDepth;
-                while (_reader.Read() && depth <= _reader.CurrentDepth)
-                {
-                }
-            }
-        }
+        public void Skip() => _reader.Skip();
```

Comment:
This is more of a general question, but why do we want to have our own copy of a JsonReader and Writer? If what we want is to actually get the document back, should we use JsonDocument.Parse and let that call the reader?

Reply:
This is lower level code that cares about perf and wants to directly use the reader/writer.